File: Swift/Utils/Utils/DeviceUtils.m
#import "DeviceUtils.h"

@implementation DeviceInfo

@end

@implementation DeviceUtils

+(DeviceInfo *) getDevieInfo {
    UIDevice * device = [UIDevice currentDevice];
    NSDictionary * dicApp = [NSBundle mainBundle].infoDictionary;
    DeviceInfo * info = [[DeviceInfo alloc] init];
    info.deviceName = device.name;
    info.uuid = device.identifierForVendor.UUIDString;
    info.system = device.systemName;
    info.systemVersion = device.systemVersion;
    info.model = device.model;
    info.appName = (NSString *)[dicApp objectForKey:BUNDLE_DISPLAY_NAME];
    info.appVersion = (NSString *)[dicApp objectForKey:BUNDLE_SHORT_VERSION];
    info.appBuild = (NSString *)[dicApp objectForKey:BUNDLE_VERSION];
    return info;
}


@end

File: Swift/Utils/DeviceUtils.swift
import UIKit

class DeviceInfo: NSObject {
    var deviceName: String?
    var uuid: String?
    var system: String?
    var systemVersion: String?
    var model: String?
    var appName: String?
    var appVersion: String?
    var appBuild: String?
}

class DeviceUtils: NSObject {
    
    class func getDevieInfo() -> DeviceInfo {
        var device = UIDevice()
        var dicApp = NSBundle.mainBundle().infoDictionary
        var info = DeviceInfo()
        info.deviceName = device.name
        info.uuid = device.identifierForVendor.UUIDString
        info.system = device.systemName
        info.systemVersion = device.systemVersion
        info.model = device.model
        info.appName = dicApp?["CFBundleDisplayName"] as? String
        info.appVersion = dicApp?["CFBundleShortVersionString"] as? String
        info.appBuild = dicApp?["CFBundleVersion"] as? String
        return info
    }
    
    
}
